testJITServer_1_FAILED Assertion failed at compiler/runtime/JITClientSession.cpp:2127: *slot == NULL || *slot == loader

Failure link
------------

From https://openj9-jenkins.osuosl.org/job/Test_openjdk24_j9_sanity.functional_ppc64le_linux_Nightly_testList_1/88/consoleFull
```
21:01:29  =JAVA VERSION OUTPUT BEGIN=
21:01:29  openjdk version "24.0.1-internal" 2025-04-15
21:01:29  OpenJDK Runtime Environment (build 24.0.1-internal-adhoc.****.BuildJDK24ppc64lelinuxNightly)
21:01:29  Eclipse OpenJ9 VM (build master-8be07bf1764, JRE 24 Linux ppc64le-64-Bit Compressed References 20250528_90 (JIT enabled, AOT enabled)
21:01:29  OpenJ9   - 8be07bf1764
21:01:29  OMR      - 3b5210dbcb4
21:01:29  JCL      - 29f868425a0 based on jdk-24.0.1+9)
```
[Rerun in Grinder](https://openj9-jenkins.osuosl.org/job/Grinder/parambuild/?SDK_RESOURCE=customized&TARGET=testList+TESTLIST=testJITServer_1&BASE_DOCKER_REGISTRY_CREDENTIAL_ID=&TEST_FLAG=&UPSTREAM_TEST_JOB_NAME=Test_openjdk24_j9_sanity.functional_ppc64le_linux_Nightly&DOCKER_REQUIRED=false&ACTIVE_NODE_TIMEOUT=&VENDOR_TEST_DIRS=&EXTRA_DOCKER_ARGS=&TKG_OWNER_BRANCH=adoptium%3Amaster&OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&PLATFORM=ppc64le_linux&GENERATE_JOBS=true&KEEP_REPORTDIR=false&PERSONAL_BUILD=false&DOCKER_REGISTRY_DIR=&ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&RERUN_ITERATIONS=0&SETUP_JCK_RUN=false&DOCKER_REGISTRY_URL_CREDENTIAL_ID=&LABEL=&EXTRA_OPTIONS=&BUILD_IDENTIFIER=Nightly&CUSTOMIZED_SDK_URL=https%3A%2F%2Fopenj9-artifactory.osuosl.org%2Fartifactory%2Fci-openj9%2FBuild_JDK24_ppc64le_linux_Nightly%2F90%2FOpenJ9-JDK24-ppc64le_linux-20250528-235728.tar.gz+https%3A%2F%2Fopenj9-artifactory.osuosl.org%2Fartifactory%2Fci-openj9%2FBuild_JDK24_ppc64le_linux_Nightly%2F90%2Ftest-images.tar.gz&JENKINS_KEY=&ADOPTOPENJDK_BRANCH=master&LIGHT_WEIGHT_CHECKOUT=false&USE_JRE=false&ARTIFACTORY_SERVER=ci-eclipse-openj9&KEEP_WORKSPACE=false&USER_CREDENTIALS_ID=&JDK_VERSION=24&DOCKER_REGISTRY_URL=&ITERATIONS=1&VENDOR_TEST_REPOS=&JDK_REPO=https%3A%2F%2Fgithub.com%2Fibmruntimes%2Fopenj9-openjdk-jdk24.git&JCK_GIT_BRANCH=master&OPENJ9_BRANCH=master&OPENJ9_SHA=8be07bf176432b9d9343183f0193dc3225b93012&JCK_GIT_REPO=&VENDOR_TEST_BRANCHES=&OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&UPSTREAM_JOB_NAME=&CLOUD_PROVIDER=&CUSTOM_TARGET=&VENDOR_TEST_SHAS=&JDK_BRANCH=openj9&LABEL_ADDITION=&ARTIFACTORY_REPO=ci-openj9&ARTIFACTORY_ROOT_DIR=Test&UPSTREAM_TEST_JOB_NUMBER=88&DOCKERIMAGE_TAG=&JDK_IMPL=openj9&TEST_TIME=&SSH_AGENT_CREDENTIAL=&AUTO_DETECT=true&SLACK_CHANNEL=&DYNAMIC_COMPILE=true&RELATED_NODES=&ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&APPLICATION_OPTIONS=&CUSTOMIZED_SDK_URL_CREDENTIAL_ID=e047f50a-26c4-4b82-8525-6da17ec68450&ARCHIVE_TEST_RESULTS=false&NUM_MACHINES=&OPENJDK_SHA=&TRSS_URL=&RERUN_FAILURE=false&USE_TESTENV_PROPERTIES=false&BUILD_LIST=functional&ADDITIONAL_ARTIFACTS_REQUIRED=&UPSTREAM_JOB_NUMBER=&STF_OWNER_BRANCH=adoptium%3Amaster&TIME_LIMIT=10&JVM_OPTIONS=&PARALLEL=None) - Change TARGET to run only the failed test targets.

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
23:04:40  Running test testJITServer_1 ...

23:04:40  variation: Mode610 -Xshareclasses:none -Xjit:optLevel=hot
23:04:40  JVM_OPTIONS:  -Xcompressedrefs -Xjit -Xgcpolicy:gencon -Xshareclasses:none -Xjit:optLevel=hot 

23:05:37  //// JITServer is ready to accept incoming requests
23:05:37  //// Assertion failed at /home/jenkins/workspace/Build_JDK24_ppc64le_linux_Nightly/openj9/runtime/compiler/runtime/JITClientSession.cpp:2127: *slot == NULL || *slot == loader
23:05:37  //// 	Unhandled exception
23:05:37  //// Type=Segmentation error vmState=0x00000000
23:05:37  //// J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
23:05:37  //// Handler1=00007FFF8D043C50 Handler2=00007FFF8D768220
23:05:37  //// R0=00007FFF8D87110C R1=00007FFF6BEE8CB0 R2=00007FFF8C7EE700 R3=00007FFF940C71B0
23:05:37  //// R4=00000000000004C4 R5=00007FFF940C71C0 R6=0000000000000000 R7=0000000000000000
23:05:37  //// R8=0000000000000000 R9=00007FFF8D8716E0 R10=00007FFF6BEEB548 R11=00007FFF6BEE8E00
23:05:37  //// R12=00007FFF8D8DB520 R13=00007FFF6BEF68E0 R14=00007FFF8D87121C R15=FFFFFFFFFFFF9010
23:05:37  //// R16=0000000000000000 R17=00007FFF940C71B0 R18=00007FFF8C58BF9D R19=00007FFF8D9CD8F0
23:05:37  //// R20=0000000000000000 R21=0000000000000000 R22=00007FFF8DA2CE18 R23=0000000000000000
23:05:37  //// R24=0000000000000BD0 R25=FFFFFFFFFFFFFFFF R26=00007FFF8C58BF90 R27=00007FFF6BEEB548
23:05:37  //// R28=0000000000000000 R29=0000000000000073 R30=000000000000000D R31=00007FFF6BEE9310
23:05:37  //// NIP=00007FFF8D8DB530 MSR=800000000280D033 ORIG_GPR3=00007FFF8D82A28C CTR=00007FFF8D8DB520
23:05:37  //// LINK=00007FFF8D872A4C XER=0000000000000000 CCR=0000000028882204 SOFTE=0000000000000001
23:05:37  //// TRAP=0000000000000300 DAR=00007FFF940C71B0 dsisr=0000000040000000 RESULT=0000000000000000
23:05:37  //// FPR0=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR1=4039000000000000 (f: 0.000000, d: 2.500000e+01)
23:05:37  //// FPR2=4055c00000000000 (f: 0.000000, d: 8.700000e+01)
23:05:37  //// FPR3=0000002e00000021 (f: 33.000000, d: 9.761181e-313)
23:05:37  //// FPR4=0000002a0000002b (f: 43.000000, d: 8.912382e-313)
23:05:37  //// FPR5=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR6=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR7=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR8=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR9=0000000000000004 (f: 4.000000, d: 1.976263e-323)
23:05:37  //// FPR10=726564616f6c203d (f: 1869357056.000000, d: 1.141143e+243)
23:05:37  //// FPR11=656c69706d6f632f (f: 1836016384.000000, d: 3.684235e+180)
23:05:37  //// FPR12=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR13=0000002000000032 (f: 50.000000, d: 6.790387e-313)
23:05:37  //// FPR14=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR15=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR16=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR17=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR18=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR19=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR20=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR21=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR22=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR23=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR24=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR25=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR26=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR27=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR28=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR29=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR30=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// FPR31=0000000000000000 (f: 0.000000, d: 0.000000e+00)
23:05:37  //// Module=/lib64/glibc-hwcaps/power10/libc.so.6
23:05:37  //// Module_base_address=00007FFF8D800000
23:05:37  //// Target=2_90_20250528_90 (Linux 5.14.0-410.el9.ppc64le)
23:05:37  //// CPU=ppc64le (8 logical CPUs) (0x3f9720000 RAM)
23:05:37  //// ----------- Stack Backtrace -----------
23:05:37  //// __strlen_power10+0x10 (0x00007FFF8D8DB530 [libc.so.6+0xdb530])
23:05:37  //// buffered_vfprintf+0xac (0x00007FFF8D87333C [libc.so.6+0x7333c])
23:05:37  //// _ZL21traceAssertionFailurePKciS0_S0_Pc+0x244 (0x00007FFF8BF8A104 [libj9jit29.so+0x78a104])
23:05:37  //// _ZN2TR15fatal_assertionEPKciS1_S1_z+0x2c (0x00007FFF8BF8BDCC [libj9jit29.so+0x78bdcc])
23:05:37  //// _ZN17ClientSessionData14setClassLoaderEPPvS0_PKc+0x54 (0x00007FFF8BE06374 [libj9jit29.so+0x606374])
23:05:37  //// _ZN17ClientSessionData25setApplicationClassLoaderEPv+0x34 (0x00007FFF8BE06424 [libj9jit29.so+0x606424])
23:05:37  //// _ZN2TR30CompilationInfoPerThreadRemote25processCompilationRequestERSt5tupleIJmjjP8J9MethodP7J9Class19TR_OptimizationPlanNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEN2J928IlGeneratorMethodDetailsTypeESt6vectorIP19TR_OpaqueClassBlockSaISH_EESJ_S1_IJ+0x2c8 (0x00007FFF8BA07D68 [libj9jit29.so+0x207d68])
23:05:37  //// _ZN2TR30CompilationInfoPerThreadRemote12processEntryER21TR_MethodToBeCompiledRN2J917J9SegmentProviderE+0xc38 (0x00007FFF8BA0A178 [libj9jit29.so+0x20a178])
23:05:37  //// _ZN2TR24CompilationInfoPerThread14processEntriesEv+0x410 (0x00007FFF8B978440 [libj9jit29.so+0x178440])
23:05:37  //// _ZN2TR24CompilationInfoPerThread3runEv+0xa8 (0x00007FFF8B978A58 [libj9jit29.so+0x178a58])
23:05:37  //// _Z30protectedCompilationThreadProcP13J9PortLibraryPN2TR24CompilationInfoPerThreadE+0xa0 (0x00007FFF8B978B10 [libj9jit29.so+0x178b10])
23:05:37  //// omrsig_protect+0x3e4 (0x00007FFF8D7696D4 [libj9prt29.so+0x396d4])
23:05:37  //// _Z21compilationThreadProcPv+0x1a8 (0x00007FFF8B979078 [libj9jit29.so+0x179078])
23:05:37  //// thread_wrapper+0x190 (0x00007FFF8DA5CC00 [libj9thr29.so+0xcc00])
23:05:37  //// start_thread+0x170 (0x00007FFF8D8A31B0 [libc.so.6+0xa31b0])
23:05:37  //// clone+0xa0 (0x00007FFF8D94D8C8 [libc.so.6+0x14d8c8])
23:05:37  //// ---------------------------------------

23:16:28  FAILED: testServer
23:16:28  java.lang.AssertionError: Expected an exit value of 0 or 143, got 255 instead.
23:16:28  	at org.testng.AssertJUnit.fail(AssertJUnit.java:59)
23:16:28  	at jit.test.jitserver.JITServerTest.destroyAndCheckProcess(JITServerTest.java:357)
23:16:28  	at jit.test.jitserver.JITServerTest.testServer(JITServerTest.java:433)

23:16:28  testJITServer_1_FAILED
```

[50x internal grinder](https://openj9-jenkins.osuosl.org/job/Grinder/4362/)

--- a/runtime/compiler/control/JITClientCompilationThread.cpp
+++ b/runtime/compiler/control/JITClientCompilationThread.cpp
@@ -443,8 +443,6 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
          ClientSessionData::VMInfo vmInfo = {};
          J9JavaVM * javaVM = vmThread->javaVM;
          vmInfo._systemClassLoader = fe->getSystemClassLoader();
-         vmInfo._extensionClassLoader = fe->getExtensionClassLoader();
-         vmInfo._applicationClassLoader = fe->getApplicationClassLoader();
          vmInfo._processID = fe->getProcessID();
          vmInfo._canMethodEnterEventBeHooked = fe->canMethodEnterEventBeHooked();
          vmInfo._canMethodExitEventBeHooked = fe->canMethodExitEventBeHooked();
@@ -3512,13 +3510,6 @@ remoteCompile(J9VMThread *vmThread, TR::Compilation *compiler, TR_ResolvedMethod
    ((TR::CompilationInfoPerThread *)compInfoPT)->clearDeserializerWasReset();
    std::vector<uintptr_t> newKnownIds = deserializer ? deserializer->getNewKnownIds(compiler) : std::vector<uintptr_t>();
 
-   // The server might have cached the VMInfo for this client before the
-   // extension and/or application class loader existed. Send the class loader
-   // pointers with the compilation request so that in that case, if they have
-   // since been created, the server will have a chance to become aware of them.
-   void *extensionClassLoader = compiler->fej9vm()->getExtensionClassLoader();
-   void *applicationClassLoader = compiler->fej9vm()->getApplicationClassLoader();
-
    // TODO: make this a synchronized region to avoid bad_alloc exceptions
    compInfo->getSequencingMonitor()->enter();
    // Collect the list of unloaded classes
@@ -3573,8 +3564,7 @@ remoteCompile(J9VMThread *vmThread, TR::Compilation *compiler, TR_ResolvedMethod
          detailsStr, details.getType(), unloadedClasses, illegalModificationList, classInfoTuple, optionsStr,
          recompMethodInfoStr, chtableUpdates.first, chtableUpdates.second, useAotCompilation,
          TR::Compiler->vm.isVMInStartupPhase(compInfoPT->getJitConfig()), aotCacheStore, aotCacheLoad, methodIndex,
-         classChainOffset, ramClassChain, uncachedRAMClasses, uncachedClassInfos, newKnownIds,
-         extensionClassLoader, applicationClassLoader
+         classChainOffset, ramClassChain, uncachedRAMClasses, uncachedClassInfos, newKnownIds
       );
 
       JITServer::MessageType response;

--- a/runtime/compiler/control/JITServerCompilationThread.cpp
+++ b/runtime/compiler/control/JITServerCompilationThread.cpp
@@ -588,8 +588,6 @@ TR::CompilationInfoPerThreadRemote::processCompilationRequest(CompilationRequest
    auto &uncachedRAMClasses      = std::get<22>(req);
    auto &uncachedClassInfos      = std::get<23>(req);
    auto &newKnownIds             = std::get<24>(req);
-   void *extensionClassLoader    = std::get<25>(req);
-   void *applicationClassLoader  = std::get<26>(req);
 
    TR_ASSERT_FATAL(TR::Compiler->persistentMemory() == compInfo->persistentMemory(),
                      "per-client persistent memory must not be set at this point");
@@ -642,8 +640,6 @@ TR::CompilationInfoPerThreadRemote::processCompilationRequest(CompilationRequest
          throw std::bad_alloc();
 
       setClientData(clientSession); // Cache the session data into CompilationInfoPerThreadRemote object
-      clientSession->setExtensionClassLoader(extensionClassLoader);
-      clientSession->setApplicationClassLoader(applicationClassLoader);
 
       // After this line, all persistent allocations are made per-client,
       // until exitPerClientAllocationRegion is called

--- a/runtime/compiler/control/JITServerCompilationThread.hpp
+++ b/runtime/compiler/control/JITServerCompilationThread.hpp
@@ -41,7 +41,7 @@ using CompilationRequest = std::tuple<
    J9::IlGeneratorMethodDetailsType, std::vector<TR_OpaqueClassBlock *>, std::vector<TR_OpaqueClassBlock *>,
    JITServerHelpers::ClassInfoTuple, std::string, std::string, std::string, std::string,
    bool, bool, bool, bool, uint32_t, uintptr_t, std::vector<J9Class *>, std::vector<J9Class *>,
-   std::vector<JITServerHelpers::ClassInfoTuple>, std::vector<uintptr_t>, void *, void *
+   std::vector<JITServerHelpers::ClassInfoTuple>, std::vector<uintptr_t>
 >;
 
 void outOfProcessCompilationEnd(TR_MethodToBeCompiled *entry, TR::Compilation *comp);

--- a/runtime/compiler/env/VMJ9.cpp
+++ b/runtime/compiler/env/VMJ9.cpp
@@ -7179,18 +7179,6 @@ TR_J9VMBase::getSystemClassLoader()
    return vmThread()->javaVM->systemClassLoader;
    }
 
-void *
-TR_J9VMBase::getExtensionClassLoader()
-   {
-   return vmThread()->javaVM->extensionClassLoader;
-   }
-
-void *
-TR_J9VMBase::getApplicationClassLoader()
-   {
-   return vmThread()->javaVM->applicationClassLoader;
-   }
-
 bool
 TR_J9VMBase::acquireClassTableMutex()
    {

--- a/runtime/compiler/env/VMJ9.h
+++ b/runtime/compiler/env/VMJ9.h
@@ -1465,8 +1465,6 @@ public:
    virtual int32_t maxInternalPlusPinningArrayPointers(TR::Compilation* comp);
 
    virtual void *getSystemClassLoader();
-   virtual void *getExtensionClassLoader();
-   virtual void *getApplicationClassLoader();
 
    virtual TR_EstimateCodeSize *getCodeEstimator( TR::Compilation *comp);
    virtual void releaseCodeEstimator( TR::Compilation *comp, TR_EstimateCodeSize *estimator);

--- a/runtime/compiler/env/VMJ9Server.cpp
+++ b/runtime/compiler/env/VMJ9Server.cpp
@@ -407,22 +407,6 @@ TR_J9ServerVM::getSystemClassLoader()
    return vmInfo->_systemClassLoader;
    }
 
-void *
-TR_J9ServerVM::getExtensionClassLoader()
-   {
-   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
-   auto *vmInfo = _compInfoPT->getClientData()->getOrCacheVMInfo(stream);
-   return vmInfo->_extensionClassLoader;
-   }
-
-void *
-TR_J9ServerVM::getApplicationClassLoader()
-   {
-   JITServer::ServerStream *stream = _compInfoPT->getMethodBeingCompiled()->_stream;
-   auto *vmInfo = _compInfoPT->getClientData()->getOrCacheVMInfo(stream);
-   return vmInfo->_applicationClassLoader;
-   }
-
 bool
 TR_J9ServerVM::jitFieldsOrStaticsAreIdentical(TR_ResolvedMethod * method1, I_32 cpIndex1, TR_ResolvedMethod * method2, I_32 cpIndex2, int32_t isStatic)
    {

--- a/runtime/compiler/env/VMJ9Server.hpp
+++ b/runtime/compiler/env/VMJ9Server.hpp
@@ -82,8 +82,6 @@ public:
    virtual TR_OpaqueClassBlock * getLeafComponentClassFromArrayClass(TR_OpaqueClassBlock * arrayClass) override;
    virtual TR_OpaqueClassBlock *getClassFromSignature(const char *sig, int32_t length, J9ConstantPool *constantPool, bool isVettedForAOT = false) override;
    virtual void * getSystemClassLoader() override;
-   virtual void * getExtensionClassLoader() override;
-   virtual void * getApplicationClassLoader() override;
    virtual bool jitFieldsAreSame(TR_ResolvedMethod * method1, I_32 cpIndex1, TR_ResolvedMethod * method2, I_32 cpIndex2, int32_t isStatic) override;
    virtual bool jitStaticsAreSame(TR_ResolvedMethod *method1, I_32 cpIndex1, TR_ResolvedMethod *method2, I_32 cpIndex2) override;
    virtual TR_OpaqueClassBlock * getComponentClassFromArrayClass(TR_OpaqueClassBlock *arrayClass) override;

--- a/runtime/compiler/net/CommunicationStream.hpp
+++ b/runtime/compiler/net/CommunicationStream.hpp
@@ -129,7 +129,7 @@ protected:
    // likely to lose an increment when merging/rebasing/etc.
    //
    static const uint8_t MAJOR_NUMBER = 1;
-   static const uint16_t MINOR_NUMBER = 81; // ID: FoxOMSLFyVA0Xn2XHmL1
+   static const uint16_t MINOR_NUMBER = 80; // ID: snCDuoE2+InCmAiOg6YU
    static const uint8_t PATCH_NUMBER = 0;
    static uint32_t CONFIGURATION_FLAGS;
 

--- a/runtime/compiler/runtime/JITClientSession.cpp
+++ b/runtime/compiler/runtime/JITClientSession.cpp
@@ -2103,35 +2103,6 @@ ClientSessionData::printSharedProfileCacheStats() const
       }
    }
 
-void
-ClientSessionData::setExtensionClassLoader(void *loader)
-   {
-   if (_vmInfo != NULL)
-      {
-      setClassLoader(&_vmInfo->_extensionClassLoader, loader, "extension");
-      }
-   }
-
-void
-ClientSessionData::setApplicationClassLoader(void *loader)
-   {
-   if (_vmInfo != NULL)
-      {
-      setClassLoader(&_vmInfo->_applicationClassLoader, loader, "application");
-      }
-   }
-
-void
-ClientSessionData::setClassLoader(void **slot, void *loader, const char *which)
-   {
-   TR_ASSERT_FATAL(
-      *slot == NULL || *slot == loader,
-      "inconsistent %s loaders: %p vs %p",
-      *slot,
-      loader);
-
-   *slot = loader;
-   }
 
 ClientSessionHT*
 ClientSessionHT::allocate()

--- a/runtime/compiler/runtime/JITClientSession.hpp
+++ b/runtime/compiler/runtime/JITClientSession.hpp
@@ -278,8 +278,6 @@ public:
    struct VMInfo
       {
       void *_systemClassLoader;
-      void *_extensionClassLoader;
-      void *_applicationClassLoader;
       uintptr_t _processID;
       bool _canMethodEnterEventBeHooked;
       bool _canMethodExitEventBeHooked;
@@ -553,11 +551,7 @@ public:
    void printSharedProfileCacheStats() const;
    void printIProfilerCacheStats();
 
-   void setExtensionClassLoader(void *loader);
-   void setApplicationClassLoader(void *loader);
-
 private:
-   void setClassLoader(void **slot, void *loader, const char *which);
    void destroyMonitors();
 
    // If this function sets the missingLoaderInfo flag then a NULL result is due to missing class loader info;

